File: algo_expert/medium/022RemoveNthNodeFromEnd/golang/remove.go
package golang

// LinkedList Data Class
type LinkedList struct {
	Value int
	Next  *LinkedList
}

//RemoveNthNodeFromEnd implementation
func RemoveNthNodeFromEnd(head *LinkedList, nth int) {
	counter, first, second := 1, head, head

	for counter <= nth {
		second = second.Next
		counter ++
	}

	if second == nil {
		head.Value = head.Next.Value
		head.Next = head.Next.Next
	}
	for second.Next != nil {
		second = second.Next
		first = first.Next
	}
	first.Next = first.Next.Next
}

File: algo_expert/medium/022RemoveNthNodeFromEnd/remove.py

class LinkedList():
    pass

def remove_nth_node_from_end(head: LinkedList, n: int):
    first = head
    second = head
    counter = 1

    while counter <= n:
        second = second.next
        counter += 1

    if second is None:
        head.value = head.next.value
        head.next = head.next.next
        return
    
    while second.next is not None:
        second = second.next
        first = first.next

    first.next = first.next.next    